addons/barcodes/models/barcode_nomenclature.py

```diff
@@ -141,7 +130,7 @@ def parse_barcode(self, barcode):
 
         rules = []
         for rule in self.rule_ids:
-            rules.append({'type': rule.type, 'encoding': rule.encoding, 'sequence': rule.sequence, 'pattern': rule.pattern, 'alias': rule.alias})
+            rules.append({'rule_type': rule.type, 'encoding': rule.encoding, 'sequence': rule.sequence, 'pattern': rule.pattern, 'alias': rule.alias})
```

Comment:
While testing barcode interface(Inside stock.picking form), scanning barcode throws traceback for type, note that type is changed to rule_type